daemon/cmld.c

```diff
@@ -204,6 +204,26 @@ cmld_is_internet_active(void)
 	return container_connectivity_online(cmld_connectivity);
 }
 
+/**
+ * Requests the SCD to initialize a token assocaited to a container and queries whether that
+ * token has been provisioned with a platform-bound authentication code.
+ */
+static int
+cmld_container_token_init(container_t *container)
+{
+	ASSERT(container);
+
+	if (smartcard_scd_token_block_add(cmld_smartcard, container) != 0) {
+		ERROR("Requesting SCD to init token failed");
+		return -1;
+	}
+
+	container_set_token_is_linked_to_device(
+		container, smartcard_container_token_is_provisioned(container));
+
```

Comment:
both functions have container as parameter -> better join them
=> e.g. smartcard_update_token_state(container) and call token_is_provisioned() inside